### PR TITLE
add -h, -b options, fix -o

### PR DESCRIPTION
**Pull request by atotic**
Added some options to make it more useful:

-h for help
-b for picking browser other than default
-o now defaults to index.html, it used to open "true".
-p if unspecified, now scans ports for free port starting at 8080

This makes my workflow a lot simpler, lots of polyserves everywhere....

### CODE REVIEW

**Comment by atotic** [comment]
Just had a hunch, and improved handling of -b. Now multiple -b options open multiple browsers at once, which is nice for testing.

    polyserve -o -b 'Chrome' -b 'Firefox' -b 'Safari' FTW!

File: src/start_server.js [hunk 1]
```diff
@@ -13,24 +13,54 @@ var http = require('http');
...
 
   var server = http.createServer(app);
-  server = app.listen(port);
 
-  var baseUrl = util.format('http://localhost:%d/components/%s/', port, polyserve.packageName);
+  server = app.listen(options.port);
+
+  server.on('error', function(err) {
+    if (err.code === 'EADDRINUSE')
+      console.error("ERROR: Port in use", options.port, "Aborting!");
```
Comment (justinfagnani) [inline src/start_server.js]: Let's replace "Aborting!" with something like "Please choose another port, let an unused port be chosen automatically"

**Comment by atotic** [comment]
Hey, ABORTING! was a huge improvement over raw error throw :)
Fixed.
PTAL

**Comment by atotic** [comment]
Fixed a license while I was in there.

File: src/start_server.js [hunk 2]
```diff
@@ -13,24 +13,54 @@ var http = require('http');
 var makeApp = require('./make_app');
 var open = require('open');
 var util = require('util');
+var findPort = require('find-port');
 
-function startServer(port, openPage, componentDir, packageName) {
-  port = port || 8080;
-  console.log('Starting Polyserve on port ' + port);
+function startWithPort(options) {
```
Comment (justinfagnani) [inline src/start_server.js]: I don't like this rename, port is just one of several options.

Comment (justinfagnani) [inline src/start_server.js]: Please document the properties available in options, now that we've lost the argument names.

File: src/start_server.js (hunk 1 above)
Comment (justinfagnani) [inline src/start_server.js]: style: wrap line to 80 cols

File: src/start_server.js [hunk 3]
```diff
@@ -13,24 +13,54 @@ var http = require('http');
...
+  var baseUrl = util.format('http://localhost:%d/components/%s/', options.port,
+    polyserve.packageName);
   console.log('Files in this directory are available under ' + baseUrl);
 
-  if (openPage) {
-    open(baseUrl + openPage);
+  if (options.page) {
+    var url = baseUrl + (options.page === true ? 'index.html' : options.page);
+    if (Array.isArray(options.browser)) {
+      for (var i=0; i<options.browser.length; i++)
```
Comment (justinfagnani) [inline src/start_server.js]: style: add spaces around `=` and `<`

File: package.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
   ],
   "dependencies": {
     "express": "^4.8.5",
+    "find-port": "^1.0.1",
```
Comment (justinfagnani) [inline package.json]: Why this change? AFAIK we always include the link to the license.

File: src/start_server.js [hunk 5]
```diff
@@ -13,24 +13,54 @@ var http = require('http');
...
+        open(url, options.browser[i]);
+    }
+    else {
+      open(url, options.browser);
+    }
+  }
+}
+
+function startServer(options) {
```
Comment (justinfagnani) [inline src/start_server.js]: ok, I see the reason for the function name above. I'd put this short function above the startWithPort for readability.

File: src/start_server.js (hunk 3 above)
Comment (atotic) [inline src/start_server.js]: that's the way I like it too, but g-w-c team has been beating me over the head about it. done,

File: src/start_server.js (hunk 2 above)
Comment (atotic) [inline src/start_server.js]: done

File: src/start_server.js (hunk 5 above)
Comment (atotic) [inline src/start_server.js]: done

File: package.json (hunk 4 above)
Comment (atotic) [inline package.json]: it's the new package.json license format. No more urls in licenses. you get a warning on install without this change.

**Comment by atotic** [comment]
Fixes checked in. PTAL.
